presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionTreeUtils.java

```diff
@@ -165,4 +170,46 @@ public static FieldId checkAndGetColumnReferenceField(Expression expression, Mul
 
         return columnReferences.get(NodeRef.of(expression)).iterator().next();
     }
+
+    public static boolean isNonNullConstant(Expression expression)
+    {
+        Expression tempExpression = expression;
```

Comment:
we don't need a new variable. Just directly use `expression`

Reply:
> we don't need a new variable. Just directly use `expression`

Hmm ok. I remember having some rule before about assigning to param. I know in java it's not an issue but generally considered not clean

Reply:
Done - keeping it this way.